.circleci/continue_config.yml

```diff
@@ -0,0 +1,249 @@
+      plugin:
+        type: string
+    machine:
+      # Don't use 2018 image: https://discuss.circleci.com/t/24639/18
+      image: circleci/classic:201711-01
+      docker_layer_caching: true
+    steps:
+      - setup:
+          python_version: <<parameters.python_version>>
+          plugin: <<parameters.plugin>>
+      - run:
+          name: Run e2e tests
+          command: make plugin=<<parameters.plugin>> e2e-tests
+
+  lint:
+    parameters:
+      plugin:
+        type: string
+    machine:
+      # Don't use 2018 image: https://discuss.circleci.com/t/24639/18
+      image: circleci/classic:201711-01
+      docker_layer_caching: true
+    steps:
+      - setup:
+          python_version: "3.8"
+          plugin: <<parameters.plugin>>
+      - run:
+          name: Run pylint and flake8
+          command: make plugin=<<parameters.plugin>> lint
+
+  win_unit_tests:
+    parameters:
+      python_version:
+        type: string
+      plugin:
+        type: string
+    executor:
+      name: win/default
+    steps:
+      - checkout
+      - win_setup_conda:
+          python_version: <<parameters.python_version>>
+      - win_setup_requirements:
+          plugin: <<parameters.plugin>>
+      - run:
+          # e2e tests are not currently runnable on CircleCI on Windows as
+          # those require the ability to run Linux containers:
+          # "The Windows executor currently only supports Windows containers.
+          # Running Linux containers on Windows is not possible for now"
+          # (from https://circleci.com/docs/2.0/hello-world-windows/)
+          name: Run unit tests
+          command: |
+            conda activate kedro_plugins
+            pytest <<parameters.plugin>>\tests
+
+workflows:
+  # when pipeline parameter, run-build-kedro-telemetry is true, the
+  # kedro-telemetry job is triggered.
+  kedro-telemetry:
+    when: <<pipeline.parameters.run-build-kedro-telemetry>>
+    jobs:
+      - unit_tests:
+          plugin: "kedro-telemetry"
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - win_unit_tests:
+          plugin: "kedro-telemetry"
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - lint:
+          plugin: "kedro-telemetry"
+  # when pipeline parameter, run-build-kedro-docker is true, the
+  # kedro-docker job is triggered.
+  kedro-docker:
+    when: <<pipeline.parameters.run-build-kedro-docker>>
+    jobs:
+      - unit_tests:
+          plugin: "kedro-docker"
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - e2e_tests:
+          plugin: "kedro-docker"
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - win_unit_tests:
+          plugin: "kedro-docker"
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - lint:
```

Comment:
Does it make sense to reorder this a bit and put the `lint` first? I believe it should be the fastest one, so if it fails, we shouldn't wait for all the tests to run to fail the whole job.

Reply:
All builds will just kick-off at the same time, so the `lint` build can still finish/fail first.